src/query/v2/plan/operator.cpp

```diff
@@ -2198,23 +2197,26 @@ void SetPropertiesOnRecord(TRecordAccessor *record, const TypedValue &rhs, SetPr
 
   auto set_props = [&, record](auto properties) {
     for (auto &kv : properties) {
-      auto maybe_error = record->SetProperty(kv.first, kv.second);
-      if (maybe_error.HasError()) {
-        switch (maybe_error.GetError()) {
-          case storage::v3::Error::DELETED_OBJECT:
-            throw QueryRuntimeException("Trying to set properties on a deleted graph element.");
-          case storage::v3::Error::SERIALIZATION_ERROR:
-            throw TransactionSerializationException();
-          case storage::v3::Error::PROPERTIES_DISABLED:
-            throw QueryRuntimeException("Can't set property because properties on edges are disabled.");
-          case storage::v3::Error::VERTEX_HAS_EDGES:
-          case storage::v3::Error::NONEXISTENT_OBJECT:
-            throw QueryRuntimeException("Unexpected error when setting properties.");
+      if constexpr (std::is_same_v<TRecordAccessor, VertexAccessor>) {
```

Comment:
Cannot you use `AccessorWithSetPropertyAndValidate<TRecordAccessor>` here? I think it is better to use that, otherwise you might use the other branch for accessors that have `SetPropertyAndValidate` functions.